youtube_dl/YoutubeDL.py

```diff
@@ -1744,32 +1744,57 @@ def process_subtitles(self, video_id, normal_subtitles, automatic_captions):
         return subs
 
     def __forced_printings(self, info_dict, filename, incomplete):
+        FIELD_ALIASES = {}
+
         def print_mandatory(field):
+            actual_field = FIELD_ALIASES.get(field, field)
             if (self.params.get('force%s' % field, False)
-                    and (not incomplete or info_dict.get(field) is not None)):
-                self.to_stdout(info_dict[field])
+                    and (not incomplete or info_dict.get(actual_field) is not None)):
+                self.to_stdout(info_dict[actual_field])
 
         def print_optional(field):
             if (self.params.get('force%s' % field, False)
                     and info_dict.get(field) is not None):
                 self.to_stdout(info_dict[field])
 
+        info_dict = info_dict.copy()
+        info_dict['duration_string'] = (  # %(duration>%H-%M-%S)s is wrong if duration > 24hrs
+            formatSeconds(info_dict['duration'])
+            if info_dict.get('duration', None) is not None
+            else None)
+        if info_dict.get('resolution') is None:
+            info_dict['resolution'] = self.format_resolution(info_dict, default=None)
+        if filename is not None:
+            info_dict['filename'] = filename
+        if info_dict.get('requested_formats') is not None:
+            # For RTMP URLs, also include the playpath
+            info_dict['urls'] = '\n'.join(f['url'] + f.get('play_path', '') for f in info_dict['requested_formats'])
+        elif 'url' in info_dict:
+            info_dict['urls'] = info_dict['url'] + info_dict.get('play_path', '')
+        if 'urls' in info_dict:
+            FIELD_ALIASES['url'] = 'urls'
+
+        for tmpl in self.params.get('forceprint', []):
+            if re.match(r'\w+$', tmpl):
+                tmpl = '%({0})s'.format(tmpl)
+            try:
+                out_txt = tmpl % info_dict
```

Comment:
One of the so far unadvertised differences from yt-dlp's `--print` is that missing or bad fields give a warning (as below) instead of the NA processing in the filename code. The implementation is just trying to address requirements like the one addressed in #30703.